src/NHibernate.TestDatabaseSetup/TestDatabaseSetup.cs

```diff
@@ -102,8 +102,17 @@ private static void SetupSqlServerOdbc(Cfg.Configuration cfg)
 
 		private static void SetupFirebird(Cfg.Configuration cfg)
 		{
+			Directory.CreateDirectory(@"D:\SqlData\Firebird");
```

Comment:
The connection string does currently only hold the alias `nhibernate`. The actual path is in Databases.conf file, in Firebird installation directory. Yes this code line is a crutch for accounting the possible loss of D: content.

When using a database server, it does not really make sens for me to locate the database relatively to the client, that is usually the server choice. Firebird allows to specify the database location from the connection string, which is a strange model for a server, not at all common. So we may do that anyway, by detecting where the test assemblies are then tweaking the connection string for telling the server where to put the file. But that would depart from what is and can be done for other actual servers like Oracle, Sql-Server, PostgreSql, which gives no option to the client about where to locate the database.
And we would have to do that both in `TestDatabaseSetup` and the test project.

Otherwise we should go back to testing Firebird in embedded mode, with all its additional binaries and apparently some new behaviors in v3 requiring a dedicated dialect. (There are seven hundred failing tests when running in embedded mode with Firebird v3, most solved by redefining the boolean type as have done Nathan. I have not yet investigated the other cases, since I do not consider we should test Firebird as an embedded server.)